src/tests/unit/test_foundry_list.py

```diff
@@ -42,9 +42,41 @@ def contracts(self) -> dict[str, Contract]:
             ret[full_method] = contract
         return ret
 
+    @cached_property
+    def all_tests(self) -> list[str]:
```

Comment:
You should set up a separate test double for the new unit test. By doing so, we can avoid having a single complicated mock for all our unit tests. (Hopefully that would solve the duplication issue as well.)

Reply:
Also, it looks like `Foundry` is pretty complicated to mock. If you have any refactoring in mind that would make testing the function simpler, feel free to do them.